src/main/java/com/parasoft/demoapp/grpc/service/JsonServiceImpl.java

```diff
@@ -1,9 +1,30 @@
 package com.parasoft.demoapp.grpc.service;
 
 
+import com.parasoft.demoapp.exception.ParameterException;
+import com.parasoft.demoapp.service.ItemInventoryService;
+import io.grpc.Status;
+import io.grpc.stub.StreamObserver;
 import net.devh.boot.grpc.server.service.GrpcService;
+import org.springframework.beans.factory.annotation.Autowired;
 
 @GrpcService
 public class JsonServiceImpl extends JsonServiceImplBase {
+    @Autowired
+    private ItemInventoryService itemInventoryService;
 
+    @Override
+    public void getStockByItemId(Long itemId, StreamObserver<Integer> responseObserver) {
+        try {
+            Integer inStock = itemInventoryService.getInStockByItemId(itemId);
```

Comment:
For service 'getInStockByItemId', it will return status with ’success' whatever the itemId does not exist or is one negtive number (And this is the existing behavior, which I just want to mention here)

Reply:
I think we should do special handling for different error case for the method since we don't have much methods in our gRPC service.
In here maybe we can check whether returned "inStock" is null and wrap a exception message like "Inventory item doesn't exist."

Reply:
Ok, I will let others know.